integration/tests/handshake.test.ts

```diff
@@ -885,3 +886,290 @@ test.describe('Client handshake @generic', () => {
+      when: {
+        initialAuthState: 'expired',
+        initialSessionClaims: new Map<string, string>([['org_id', 'org_a']]),
+        orgSyncOptions: {
+          organizationPatterns: ['/organizations-by-id/:id', '/organizations-by-id/:id/*splat'],
+        },
+        appRequestPath: '/',
+      },
+      then: {
+        expectStatus: 307,
+        fapiOrganizationIdParamValue: null,
+      },
+    },
+    {
+      name: 'When org A is active but org B is requested by ID, attempts to activate org B (active)',
+      when: {
+        initialAuthState: 'active',
+        initialSessionClaims: new Map<string, string>([['org_id', 'org_a']]),
+        orgSyncOptions: {
+          organizationPatterns: ['/organizations-by-id/:id'],
+          personalWorkspacePatterns: ['/personal-workspace'], // <-- Unnecessary
+        },
+        appRequestPath: '/organizations-by-id/org_b',
+      },
+      then: {
+        expectStatus: 307,
+        fapiOrganizationIdParamValue: 'org_b',
+      },
+    },
+    {
+      name: 'When the personal workspace is active but org A is requested by ID, attempts to activate org A',
+      when: {
+        initialAuthState: 'active',
+        initialSessionClaims: new Map<string, string>([
+          // Intentionally no org claims - means personal workspace
+        ]),
+        orgSyncOptions: {
+          organizationPatterns: ['/organizations-by-id/:id'],
+          personalWorkspacePatterns: ['/personal-workspace'],
+        },
+        appRequestPath: '/organizations-by-id/org_a',
+      },
+      then: {
+        expectStatus: 307,
+        fapiOrganizationIdParamValue: 'org_a',
+      },
+    },
+    {
+      name: 'When org A is active but the personal workspace is requested, attempt to activate the personal workspace',
+      when: {
+        initialAuthState: 'active',
+        initialSessionClaims: new Map<string, string>([['org_id', 'org_a']]),
+        orgSyncOptions: {
+          organizationPatterns: ['/organizations-by-id/:id'],
+          personalWorkspacePatterns: ['/personal-workspace'],
+        },
+        appRequestPath: '/personal-workspace',
+      },
+      then: {
+        expectStatus: 307,
+        fapiOrganizationIdParamValue: '',
+      },
+    },
+    {
+      name: 'Activates nothing with a broken path pattern',
+      when: {
+        initialAuthState: 'active',
+        initialSessionClaims: new Map<string, string>([['org_id', 'org_a']]),
+        orgSyncOptions: {
+          organizationPatterns: ['i am not a valid path pattern'],
+          personalWorkspacePatterns: ['And neither am I!'],
+        },
+        appRequestPath: '/personal-workspace',
+      },
+      then: {
+        expectStatus: 200,
+        fapiOrganizationIdParamValue: null,
+      },
+    },
+  ];
+
+  for (const testCase of cookieAuthCases) {
+    ['Cookie', 'Authorization'].forEach(authHeader => {
```

Comment:
Note: This means we have ~20 test cases (10 for each authorization type), which takes about 1.5 mins to execute on my machine in parallel.

I don't feel like we _really_ need all of these test cases for both Cookie and Authorization, because in the actual implementation the logic is all shared. So we could get away with just doing one or two test cases for one of these auth types if we feel like we need to get the test runtime down.